include/message_filters/subscriber.h

```diff
@@ -56,7 +57,12 @@ class SubscriberBase
    * \param topic The topic to subscribe to.
    * \param qos (optional) The rmw qos profile to use to subscribe
    */
-  virtual void subscribe(rclcpp::Node::SharedPtr node, const std::string& topic, const rmw_qos_profile_t qos = rmw_qos_profile_default) = 0;
+  virtual void subscribe(
+    rclcpp::Node::SharedPtr node,
+    const std::string& topic,
+    const rmw_qos_profile_t qos = rmw_qos_profile_default,
+    rclcpp::SubscriptionOptionsWithAllocator<AllocatorT> options =
+      rclcpp::SubscriptionOptionsWithAllocator<AllocatorT>()) = 0;
```

Comment:
It would be nice if we could introduce these options without breaking API, but I'm not sure if that's possible in a pure virtual class.

To be honest, it's not clear to me why we have this abstract class to begin with, it doesn't appear to be used anywhere (besides being implemented by `Subscriber` below) :thinking:  It might be safer to omit the changes to `SubscriberBase` and add new overloads to `Subscriber` to incorporate the subscriber options.

Pinging @mjcarroll for thoughts (I think you're the maintainer?).

Reply:
> To be honest, it's not clear to me why we have this abstract class to begin with

It doesn't seem to be meant to be extended by external users, so it might be ok to just add new virtual methods to it.

Reply:
If it's not meant to be extended, then I think we could just add the new method to the concrete class. 

We could go so far as to deprecate the pure virtual class and/or remove it in a separate PR, to avoid similar API/ABI issues in the future.

Reply:
I'm also okay with your proposal to implement and warn about the non-overridden method https://github.com/ros2/message_filters/pull/53#discussion_r626682459